lib/resque/worker.rb

```diff
@@ -180,6 +180,52 @@ def glob_match(pattern)
       end.sort
     end
 
+    def kill_key
+      "worker:#{self}:kill"
+    end
+
+    def terminal?
+      redis.exists(kill_key)
+    end
+
+    def retry?
+      redis.get(kill_key) == "retry"
+    end
+
+    def kill!
+      redis.setex(kill_key, 300, "kill")
```

Comment:
The UI talks about killing killing a job or its attempt, but we don't store an identifier for the job so that we don't accidentally kill the wrong job.  We intended to do that in Shopify, but this initial version never did that.